### PR TITLE
Allow custom my.cnf for mariadb from project .ddev, for drud/ddev#654

### PR DESCRIPTION
**Pull request by rfay**
## The Problem:

There are just going to be times when users have to override the mariadb config. OP drud/ddev#654 is an example where the database just requires it. 

## The Fix:

ddev will now mount the project's .ddev into /mnt/ddev_config. If there's a /mnt/ddev_config/mysql has any cnf files on there, then mysqld will include them.

A temporary push of this container is at `drud/mariadb-local:20180308_allow_config_override `

## The Test:

Manual testing is best done by going through the process using ddev. https://github.com/drud/ddev/pull/707

## Automation Overview:

* Added tests to make sure that the config can be overridden.

## Related Issue Link(s):

* ddev version bumper and related code: https://github.com/drud/ddev/pull/707
* OP drud/ddev#654

### CODE REVIEW

**Comment by rfay** [comment]
@cweagans pointed out that an override would be better than requiring the user to repeat the whole file, for both php and mysql. I'll change this to work that way.

File: test/testserver.sh [hunk 1]
```diff
@@ -99,5 +100,27 @@ if [ ! -f $outdir/mariadb_10.1_base_db.tgz ] ; then
...
+
+if ! containercheck; then
+	echo "Container did not become ready"
+fi
+
+# Make sure the custom config is present in the container.
+docker exec -it $CONTAINER_NAME grep "collation-server" /ddev_config/mysql/utf.cnf
+
+# Test that includedir is present in my.cnf
+docker exec -it $CONTAINER_NAME grep "includedir" /etc/my.cnf
```
Comment (rfay) [inline test/testserver.sh]: Let's make this a bigger grep, maybe includedir.*ddev - in case a future my.cnf has another include.

**Comment by rfay** [comment]
Minor changes per comments. I ended up making sure the config directory was there instead of doing the possibly fragile edit of my.cnf. Moved the mount to the more logical /mnt/ddev_config. Would appreciate one last look @cweagans at 9ac3688

Pushed this image to drud/mariadb-local:20180308_allow_config_override

**Comment by tmotyl** [comment]
cross linking documentation - how to apply custom mysql config for ddev
https://ddev.readthedocs.io/en/latest/users/extend/customization-extendibility/#providing-custom-mysqlmariadb-configuration-mycnf

**Pull request merged**